File: Iss_blood_front/src/main/java/Validators/RegisterValidator.java
package Validators;

import Model.RegisterInfo;
import javafx.util.Pair;

public class RegisterValidator extends Validator<RegisterInfo> {

    /**
     * @param info - the RegisterInfo to be validated
     * @return True if the info is valid, false otherwise
     */
    public Pair<Boolean, String> validate(RegisterInfo info)
    {

        String erori = "";

        if(info.getAddress().equals("") || info.getEmail().equals("") || info.getNume().equals("") ||
                info.getPassword().equals("") || info.getPhone().equals("") || info.getUsername().equals(""))
            erori += "Toate campurile sunt obligatorii\n";

        if(info.getPhone().length() != 10)
            erori += "Numarul de telefon trebuie a aiba 10 cifre\n";

        if(!checkForDigits(info.getPhone()))
            erori += "Numarul de telefon nu e valid\n";

        if(info.getCnp().length() != 13)
            erori += "CNP-ul trebuie sa aiba 13 cifre\n";

        if(!checkForDigits(info.getCnp()))
            erori += "CNP-ul nu e valid\n";

        if (erori.equals(""))
            return new Pair<>(true, "Ok");
        return new Pair<>(false, erori);
    }
}

File: Iss_blood_back/Validators/register_validator.py
from Model.account_type import AccountType
from Utils.user_utils import data_nasterii_din_cnp


class register_validator:
    def validate(self, register_info):
        '''
        Verifica daca informatia data e valida
        O parte din validari se intampla si in service(e nevoie de baza de date)
        :param register_info:
        :return: Tuple>bool, string) -> valid/invalid + mesaj cu eventualele erori
        '''

        erori = ""

        if register_info.username == "" or register_info.password == "" or register_info.email == "" or \
                register_info.nume == "" or register_info.prenume == "" or register_info.cnp == "" or \
                register_info.judet == "" or register_info.localitate == "" or register_info.address == "" or \
                register_info.phone == "" or register_info.account_type == "": \
                erori += "Toate campurile sunt obligatorii\n"

        if register_info.account_type is None:
            erori += "Eoare interna: tipul de cont nu a fost gasit\n"
        elif register_info.account_type != AccountType.Donator and register_info.license == "":
            # verificare separata: donatorii nu au licenta
            erori += "Toate campurile sunt obligatorii\n"

        if len(register_info.phone) != 10:
            erori += "Numarul de telefon nu e valid\n"
        else:
            for c in register_info.phone:
                if c < '0' or c > '9':
                    erori += "Numarul de telefon nu e valid\n"

        if "@" not in register_info.email or "." not in register_info.email:
            erori += "Adresa de email nu e valida\n"

        try:
            register_info.data_nasterii = data_nasterii_din_cnp(register_info.cnp)
        except ValueError:
            erori += "CNP-ul nu e valid\n"

        if erori == "":
            return True, ""
        return False, erori
